plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1426,6 +1429,10 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
             transactionLogWriter.flush();
             writeCommitted = true;
 
+            if (handle.replace() && readVersion != commitVersion) {
```

Comment:
Why checking `readVersion != commitVersion` ?

I'm guessing that we'll have for existing tables `handle.readVersion()` always smaller than `commitVersion`. 
Or am I missing something?

Reply:
This was so that we wouldn't write the checkpoint upon creating a new table, as `handle.replace()` is true in this scenario (e.g. [here](https://github.com/trinodb/trino/pull/21609/files#diff-1e47ed9486d59daf7476d29553faf526f23fa66450dc3a9f62d43187ed1de322R155-R164)). But you're right that we can make this more precise by changing `!=` with `<`, on it!